rotkehlchen/tests/api/test_evm_transactions.py

```diff
@@ -89,3 +89,48 @@ def test_query_transactions(rotkehlchen_api_server: 'APIServer'):
         for entry in result['entries']:
             assert entry['entry']['timestamp'] <= last_ts
             last_ts = entry['entry']['timestamp']
+
+
+@pytest.mark.parametrize('optimism_accounts', [[
+    '0xb8553D9ee35dd23BB96fbd679E651B929821969B',
+]])
+@pytest.mark.parametrize('should_mock_price_queries', [True])
+@pytest.mark.parametrize('default_mock_price_value', [FVal(1.5)])
+@pytest.mark.parametrize('start_with_valid_premium', [True])
+def test_query_transactions_single_chain(rotkehlchen_api_server: 'APIServer'):
+    """
+    Test that querying the evm transactions endpoint for transactions in a single evm chain
+    works even if the address has transactions in other chains too.
+
+    This test uses real data.
+
+    TODO: Mock network here. Need to mock transaction query for both mainnet and optimism etherscan
+    """
+    # Ask for all evm transactions (test addy has both optimism and mainnet)
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'evmtransactionsresource',
+        ), json={
+            'accounts': [{
+                'address': '0xb8553D9ee35dd23BB96fbd679E651B929821969B',
+                'evm_chain': 'optimism',
+            }],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    expected_file = Path(__file__).resolve().parent.parent / 'data' / 'expected' / 'test_evm_transactions-test_query_transactions.json'  # noqa: E501
+    with open(expected_file) as f:
+        expected_data = json.load(f)
+
+    # check all expected data exists. User has done more transactions since then if we don't
+    # mock network, so we need to test like this
+    last_ts = result['entries'][0]['entry']['timestamp']
+    for entry in expected_data['entries']:
+        if entry['entry']['evm_chain'] != 'optimism':
+            continue
+        assert entry in result['entries']
+        assert entry['entry']['timestamp'] <= last_ts
+        last_ts = entry['entry']['timestamp']
```

Comment:
isn't this already tested in `test_query_transactions`?

Reply:
Kinda yeah ... so perhaps is not needed here :+1: